.instrumentation_generator/instrumentation_generator.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require_relative '../api/lib/opentelemetry/version'
+require 'thor'
+
+class InstrumentationGenerator < Thor::Group
+  include Thor::Actions
+
+  source_root File.dirname(__FILE__)
+  argument :instrumentation_name
+
+  def root_files
+    template('templates/rubocop.yml.tt', "#{instrumentation_path}/.rubocop.yml")
+    template('templates/yardopts.tt', "#{instrumentation_path}/.yardopts")
+    template('templates/Appraisals', "#{instrumentation_path}/Appraisal")
+    template('templates/CHANGELOG.md.tt', "#{instrumentation_path}/CHANGELOG.md")
+    template('templates/Gemfile', "#{instrumentation_path}/Gemfile")
+    template('templates/LICENSE', "#{instrumentation_path}/LICENSE")
+    template('templates/gemspec.tt', "#{instrumentation_path}/#{instrumentation_gem_name}.gemspec")
+    template('templates/Rakefile', "#{instrumentation_path}/Rakefile")
+    template('templates/Readme.md.tt', "#{instrumentation_path}/README.md")
+  end
+
+  def lib_files
+    template('templates/lib/entrypoint.rb', "#{instrumentation_path}/lib/#{instrumentation_gem_name}.rb")
+    template('templates/lib/instrumentation.rb.tt', "#{instrumentation_path}/lib/opentelemetry/instrumentation.rb")
+    template('templates/lib/instrumentation/instrumentation_name.rb.tt', "#{instrumentation_path}/lib/opentelemetry/instrumentation/#{instrumentation_name}.rb")
+    template('templates/lib/instrumentation/instrumentation_name/instrumentation.rb.tt', "#{instrumentation_path}/lib/opentelemetry/instrumentation/#{instrumentation_name}/instrumentation.rb")
+    template('templates/lib/instrumentation/instrumentation_name/version.rb.tt', "#{instrumentation_path}/lib/opentelemetry/instrumentation/#{instrumentation_name}/version.rb")
+  end
+
+  def test_files
+    template('templates/test/test_helper.rb', "#{instrumentation_path}/test/test_helper.rb")
+    template('templates/test/instrumentation.rb', "#{instrumentation_path}/test/#{instrumentation_path}/instrumentation_test.rb")
+  end
+
+  def add_to_releases
+    release_details = <<-HEREDOC
+  - name: #{instrumentation_gem_name}
+    directory: #{instrumentation_path}
+    version_constant: [OpenTelemetry, Instrumentation, #{pascal_cased_instrumentation_name}, VERSION]\n
+    HEREDOC
+
+    insert_into_file('.toys/.data/releases.yml', release_details, after: "gems:\n")
+  end
+
+  def add_to_instrumentation_all
+    instrumentation_all_path = 'instrumentation/all'
+    gemfile_text = "\ngem '#{instrumentation_gem_name}', path: '../#{instrumentation_name}'"
+    insert_into_file("#{instrumentation_all_path}/Gemfile", gemfile_text, after: "gemspec\n")
+
+    gemspec_text = "\n  spec.add_dependency '#{instrumentation_gem_name}', '~> #{opentelemetry_version}'"
```

Comment:
Maybe just hardcode the version to 0.0.0 here? Or do you want to assume the AmazingGem version is lockstepped to the the otel version?

Reply:
The reason I did that is because we've been moving all the versions in lockstep so far, (excluding last nights `0.10.1` release).  Until we decide we settle on how we're going to be doing all the versioning it might make sense to keep this generator assuming lockstep.